salt/minion.py

```diff
@@ -587,8 +587,21 @@ def eval_master(self,
                 opts.update(prep_ip_port(opts))
                 opts.update(resolve_dns(opts))
                 try:
-                    pub_channel = salt.transport.client.AsyncPubChannel.factory(self.opts, **factory_kwargs)
-                    yield pub_channel.connect()
+                    if self.opts['transport'] == 'detect':
+                        detect_connect = False
+                        self.opts['detect_mode'] = True
```

Comment:
Am I right that `detect_mode` option is a flag meaning _"currently we're detecting the actual transport"_. Shouldn't it be set to `False` or unset after we're successfully connected?